### PR TITLE
Json file uploader

### PR DESCRIPTION
**Pull request by david733**
A number of changes to allow loading and uploading of .json files. Integrates Ace Editor to preview the JSON contents.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -42,16 +45,12 @@
   <script src="./src/converters.js"></script>
   <link rel="stylesheet" type="text/css" href="./src/fluxApp.css">
 
-  <!-- Google Analytics -->
```
Comment (kylegmaxwell) [inline index.html:45]: Why did you remove this, are we using different analytics now?

File: lib/ace/ext-beautify.js [hunk 2]
```diff
@@ -0,0 +1,334 @@
+ace.define("ace/ext/beautify/php_rules",["require","exports","module","ace/token_iterator"], function(require, exports, module) {
```
Comment (kylegmaxwell) [inline lib/ace/ext-beautify.js]: Are you actually using all these extensions? I would prefer to see just the ace editor minified file and maybe a couple config files, not 27 new files.

File: src/converters.js [hunk 3]
```diff
@@ -215,6 +215,22 @@ function convertCsv(inCsv) {
   });
 }
 
+function convertJson(inJson) {
+  // TODO: provide more useful response to user if JSON.parse fails. Should tell them there was a problem
+  // with the JSON, and ideally, the location. Right now the app just "hangs" with preview loading animation.
+  return new Promise(function(resolve, reject) {
+    try {
+      let json = JSON.parse(inJson);
```
Comment (kylegmaxwell) [inline src/converters.js]: Don't use let unless you are using babel

File: src/converters.js [hunk 4]
```diff
@@ -215,6 +215,22 @@ function convertCsv(inCsv) {
...
+      resolve({
+        fluxData: json
+      });
+    } catch (e) {
```
Comment (kylegmaxwell) [inline src/converters.js]: I don't think you need try catch here. The exception will automatically cause the promise to reject.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,8 @@
 # flux-uploader
 Flux SDK app that can upload to data keys and visualize geometry.
+
+To run locally:
+1. From command line in the flux-uploader folder, run npm install express
```
Comment (kylegmaxwell) [inline README.md]: You can just say 'npm install', not 'npm install express', express is already specified as a dependency in package.json.

**Review by kylegmaxwell — CHANGES_REQUESTED** [review]
ok, nice work, this is working, but just a few small things to fix.

File: index.html (hunk 1 above)
Comment (david733) [reply index.html:45]: Yes, this was a requested change by Javier.

File: lib/ace/ext-beautify.js (hunk 2 above)
Comment (david733) [reply lib/ace/ext-beautify.js]: Removed all but one of the extensions, replaced remaining with minified versions.

File: src/converters.js (hunk 3 above)
Comment (david733) [reply src/converters.js]: fixed

File: src/converters.js (hunk 4 above)
Comment (david733) [reply src/converters.js]: removed try catch

File: README.md (hunk 5 above)
Comment (david733) [reply README.md]: fixed

**Comment by david733** [comment]
Thanks Kyle! Merging to master.

**Pull request merged**